sdk/typescript/src/client/ika-client.ts

```diff
@@ -138,6 +221,100 @@ export class IkaClient {
+			return Array.from(this.cachedEncryptionKeys.values());
+		}
+
+		await this.#fetchEncryptionKeys();
+		return Array.from(this.cachedEncryptionKeys.values());
+	}
+
+	/**
+	 * Get the latest network encryption key.
+	 * This is the most recent encryption key created for the network.
+	 *
+	 * @returns Promise resolving to the latest network encryption key
+	 * @throws {NetworkError} If the encryption keys cannot be fetched
+	 */
+	async getLatestNetworkEncryptionKey(): Promise<NetworkEncryptionKey> {
+		const keys = await this.getAllNetworkEncryptionKeys();
+		if (keys.length === 0) {
+			throw new NetworkError('No network encryption keys found');
+		}
+		return keys[keys.length - 1];
+	}
+
+	/**
+	 * Get a specific network encryption key by ID.
+	 *
+	 * @param encryptionKeyID - The ID of the encryption key to retrieve
+	 * @returns Promise resolving to the specified network encryption key
+	 * @throws {ObjectNotFoundError} If the encryption key is not found
+	 * @throws {NetworkError} If the encryption keys cannot be fetched
+	 */
+	async getNetworkEncryptionKey(encryptionKeyID: string): Promise<NetworkEncryptionKey> {
+		const keys = await this.getAllNetworkEncryptionKeys();
+		const key = keys.find((k) => k.id === encryptionKeyID);
+		if (!key) {
+			throw new ObjectNotFoundError(`Network encryption key ${encryptionKeyID} not found`);
+		}
+		return key;
+	}
+
+	/**
+	 * Get the network encryption key used by a specific dWallet.
+	 * This method automatically detects which encryption key the dWallet uses.
+	 *
+	 * @param dwalletID - The ID of the dWallet to check
+	 * @returns Promise resolving to the network encryption key used by the dWallet
+	 * @throws {InvalidObjectError} If the dWallet cannot be parsed
+	 * @throws {NetworkError} If the network request fails
+	 */
+	async getDWalletNetworkEncryptionKey(dwalletID: string): Promise<NetworkEncryptionKey> {
+		const dWallet = await this.getDWallet(dwalletID);
+
+		const encryptionKeyID = dWallet.dwallet_network_encryption_key_id;
+
+		return this.getNetworkEncryptionKey(encryptionKeyID);
+	}
+
+	/**
+	 * Get the network encryption key based on client configuration.
+	 * This method respects the client's encryption key options.
+	 *
+	 * @returns Promise resolving to the appropriate network encryption key
+	 * @throws {NetworkError} If the encryption keys cannot be fetched
+	 */
+	async getConfiguredNetworkEncryptionKey(): Promise<NetworkEncryptionKey> {
+		if (this.encryptionKeyOptions.encryptionKeyID) {
+			// Use specific encryption key if configured
+			return this.getNetworkEncryptionKey(this.encryptionKeyOptions.encryptionKeyID);
+		} else if (this.encryptionKeyOptions.useLatest === false) {
+			// If useLatest is explicitly false but no specific key provided, throw error
+			throw new Error('Must specify encryptionKeyID when useLatest is false');
+		} else {
+			// Default to latest encryption key
+			return this.getLatestNetworkEncryptionKey();
+		}
```

Comment:
I don't think we need `useLatest`. Just, if the client doesn't want a particular key, use the latest, no need for a boolean